### PR TITLE
Relax version requirement for Stoplight dependency to include 3.0

### PR DESCRIPTION
**Pull request by pragmaticsc**
Stoplight relieved a redis pipeline deprecation in 3.0 that is very noisy. We are unable to upgrade our stoplight dependency until this version is relaxed.

https://github.com/bolshakov/stoplight/releases/tag/v3.0.0

### CODE REVIEW

**Review by jchengj — APPROVED** [review]

File: faraday_middleware-circuit_breaker.gemspec [hunk 1]
```diff
@@ -28,7 +28,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.add_dependency 'faraday', '>= 0.9', '< 2.0'
-  spec.add_dependency 'stoplight', '~> 2.1'
+  spec.add_dependency 'stoplight', '> 2.1', '< 4.0'
```
Comment (gottfrois) [inline faraday_middleware-circuit_breaker.gemspec]: Shouldn't it be `>= 2.0, < 4.0`?

File: lib/faraday_middleware/circuit_breaker/version.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module FaradayMiddleware
   module CircuitBreaker
-    VERSION = "0.4.1"
+    VERSION = "0.4.2"
```
Comment (gottfrois) [inline lib/faraday_middleware/circuit_breaker/version.rb]: Can you leave the version change and the changelog out of the PR? I usually do them on separate PRs :)

File: faraday_middleware-circuit_breaker.gemspec (hunk 1 above)
Comment (timbogit) [reply faraday_middleware-circuit_breaker.gemspec]: I think it should be `'>= 2.1', '< 4.0'`, as you *do* want to allow stoplight 2.1 and above (including 2.1), but all versions 3.x, and *not* (yet) 4.x.
Right @gottfrois / @vandux ?

Comment (pragmaticsc) [reply faraday_middleware-circuit_breaker.gemspec]: 👍

**Comment by pragmaticsc** [comment]
@gottfrois thank you for the prompt feedback 💪
